### PR TITLE
CVSB-417 - CVSB-949 - CVSB-1140 - Added automation for test result an…

### PR DESCRIPTION
**Pull request by lgvasile**
…d major refactoring

### CODE REVIEW

File: src/main/java/util/WriterReader.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+        FileUtils p1 = new FileUtils(WebDriverBrowsertack.getToken());
+
+        try {
+            FileOutputStream f = new FileOutputStream(new File(FILE_NAME));
+            ObjectOutputStream o = new ObjectOutputStream(f);
+
+            o.writeObject(p1);
+
+            o.close();
```
Comment (dorinNicolae) [inline src/main/java/util/WriterReader.java]: put the closing stream sequence in a finally block.

Comment (lgvasile) [reply src/main/java/util/WriterReader.java]: moving object creation in twr block, and deleting close calls

File: src/main/java/model/testresults/VehicleClass.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "VehicleClass{" +
+                "code='" + code + '\'' +
+                ", description='" + description + '\'' +
+                '}';
+    }
+}
```
Comment (teoAlexandru) [inline src/main/java/model/testresults/VehicleClass.java]: newline

File: src/main/java/util/WebDriverBrowsertack.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+
+        WebElement element4 = driver.findElement(By.xpath("//div[@class='inline-block'][1]"));
+        element4.click();
+
+        String token = driver.getCurrentUrl().split("id_token=")[1].split("&session_state=")[0];
+        driver.quit();
+        return token;
+    }
+}
```
Comment (teoAlexandru) [inline src/main/java/util/WebDriverBrowsertack.java]: newline

File: src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java [hunk 4]
```diff
@@ -0,0 +1,898 @@
...
+        testResultsSteps.postTestResults(vehicleCancelledData.setVin(generateRandomExcludingValues(21, vehicleCancelledData.build().getVin()))
+                .setVrm(generateRandomExcludingValues(7, vehicleCancelledData.build().getVrm()))
+                .setVehicleSize("small").build());
+
+        testResultsSteps.statusCodeShouldBe(201);
+        testResultsSteps.validateData("Test records created");
+        validateSavedData();
+    }
+
+}
```
Comment (teoAlexandru) [inline src/test/java/testresults/PostTestResultsPozMainLvlCancelled.java]: newline

File: src/test/java/testresults/PostTestResultsPozTestTypesCancelledLvl.java [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+                .getTestTypes().get(0).setAdditionalCommentsForAbandon("");
+
+        testResultsSteps.postTestResults(vehicleCancelledData.build());
+        testResultsSteps.statusCodeShouldBe(201);
+        testResultsSteps.validateData("Test records created");
+        validateSavedData();
+    }
+
+
+}
```
Comment (teoAlexandru) [inline src/test/java/testresults/PostTestResultsPozTestTypesCancelledLvl.java]: newline

**Pull request merged**